pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -99,9 +98,7 @@ def advance(self, *args: Any, **kwargs: Any) -> None:
         """Performs evaluation on one single dataloader"""
         void(*args, **kwargs)
         dataloader = self.trainer.accelerator.process_dataloader(self.current_dataloader)
-        data_fetcher = DataFetcher()
-        data_fetcher.setup(dataloader)
-        dataloader_iter = enumerate(data_fetcher)
+        dataloader_iter = self.trainer.data_connector.get_profiled_dataloader(dataloader)
```

Comment:
nit: this is still the dataloader right? should the var name continue as `dataloader` to avoid confusion as to whether it's the dataloader or the iterator over the dataloader?

Reply:
This is actually a `dataloader_iter` as prefetcher_iter = iter(self.data_fetcher) is being called.

```py
    def get_profiled_dataloader(self, dataloader: Iterable) -> Iterable:
        stage: str = self.trainer.state.stage.value
        self.data_fetcher = self._select_data_fetcher()
        self.data_fetcher.setup(
            dataloader,
            stage=stage,
            batch_to_device=self.trainer.accelerator.batch_to_device,
            profiler=self.trainer.profiler,
        )
        prefetcher_iter = iter(self.data_fetcher)
        return enumerate(prefetcher_iter)
```